### PR TITLE
Reuse subject set number for counts when possible

### PR DESCRIPTION
**Pull request by forus**

### CODE REVIEW

File: src/app/services/count.service.spec.ts [hunk 1]
```diff
@@ -43,4 +44,39 @@ describe('CountService', () => {
         done();
       });
   });
+
+  it('should request counts if non subject set aware constraint', () => {
+    let resourceServiceCounts = spyOn(countService['resourceService'], 'getCounts');
```
Comment (gijskant) [inline src/app/services/count.service.spec.ts]: You can also specify `resourceService = TestBed.get(ResourceService);` in the `beforeEach` block, just as for the count service, and `spyOn(resourceService, 'getCounts')`.

File: src/app/services/count.service.spec.ts [hunk 2]
```diff
@@ -43,4 +44,39 @@ describe('CountService', () => {
...
+    countService.showObservationCounts = false;
+    let subjectSetConstraint = new SubjectSetConstraint();
+    subjectSetConstraint.setSize = 10;
+
+    countService.getCounts(subjectSetConstraint).subscribe((counts) => {
+      expect(counts.subjectCount).toEqual(10);
+      expect(counts.observationCount).toEqual(-1);
+    });
+
+    expect(resourceServiceCounts).toHaveBeenCalledTimes(0);
```
Comment (gijskant) [inline src/app/services/count.service.spec.ts]: Or, alternatively, `expect(resourceServiceCounts).not.toHaveBeenCalled();`.

**Comment by gijskant** [comment]
Locally, you can run `yarn lint` to perform the linting check that is currently failing. The linter rules enforce one type of quotes using in imports: the single quote. We should probably edit the `.editorconfig` file to reflect this rule.

File: src/app/services/count.service.ts [hunk 3]
```diff
@@ -116,6 +115,21 @@ export class CountService {
     });
   }
 
+  /**
+   * When no observation counts requested, takes advantage of subject set constraint knowing number of subjects.
+   * Falls back to making reqular call otherwise
+   * @param constraint
+   */
+  getCounts(constraint: Constraint) {
```
Comment (ewelinagr) [inline src/app/services/count.service.ts]: I am still thinking whether it is better to keep this logic here or move it to transmart-resource.service, but maybe for now it is fine like this and later we will find some better place for all the optimization-related logic.

**Comment by forus** [comment]
@ewelinagr @gijskant Thanks, I've just pushed the changes.

**Pull request merged**